src/main/webapp/utilities/EventManager.js

```diff
@@ -0,0 +1,118 @@
+export class EventManager {
+    /**
+     * Constructor for normal usage
+     * @constructor
+     */
+    constructor() {
+        this.localStorage = Window.localStorage;
+
+        setup();
+    }
+
+    /**
+     * Constructor that enables injecting a mock object (for testing)
+     * @constructor
+     * @param {*} LocalStorageObject the mock object
+     */
+    constructor(LocalStorageObject) {
+        this.localStorage = localStorageObject;
+
+        setup();
+    }
+
+    /**
+     * Gets JSON-parsed value from LocalStorage
+     * @returns {Any} the value on that field
+     */
+    getValueInLocalStorageParsed() {
+        return JSON.parse(this.localStorage.getItem('events'));
+    }
+
+    /**
+     * Sets value to LocalStorage
+     * @param {Any} value the value to be set (if not string -> convert to one using JSON)
+     * @returns {void}
+     */
+    setValueInLocalStorage(value) {
+        if(typeof value === 'string') {
+            this.localStorage.setItem('events', value);
+        } else {
+            const jsonSerialized = JSON.stringify(value);
+
+            this.localStorage.setItem('events', jsonSerialized);
+        }
+    }
+
+    /**
+     * Initial setup of the LocalStorage
+     * @returns {void}
+     */
+    setup() {
+        this.eventsList = this.getValueInLocalStorageParsed();
+
+        if(events === null) { // no events object exists - first time users
```

Comment:
I don't see this variable used anywhere else. Is it correct?

Reply:
That was a mistake! Fixed it!